src/DataManagement/Saving/AlternativeSavingCalculator.cs

```diff
@@ -8,47 +8,75 @@ namespace ePiggyWeb.DataManagement.Saving
 
             for (var i = enumCount; i > (int)Importance.Necessary; i--)
             {
-                var expenses = entryList.GetBy((Importance)i - 1);
+                var expenses = entryList.GetBy((Importance)i);
                 var ratio = enumCount - i;
                 foreach (var entry in expenses)
                 {
                     var amountAfterSaving = savingType switch
                     {
-                        SavingType.Minimal => entry.Amount * SavingRatio * ratio * MinimalSavingValue,
-                        SavingType.Regular => entry.Amount * SavingRatio * ratio * RegularSavingValue,
-                        SavingType.Maximal => entry.Amount * SavingRatio * ratio * MaximalSavingValue,
+                        SavingType.Minimal => entry.Amount * ratio * MinimalSavingValue,
+                        SavingType.Regular => entry.Amount * ratio * RegularSavingValue,
+                        SavingType.Maximal => entry.Amount * ratio * MaximalSavingValue,
                         _ => throw new ArgumentOutOfRangeException()
                     };
                     entrySuggestions.Add(new SavingSuggestion(entry, amountAfterSaving));
 
-                    startingBalance += entry.Amount - amountAfterSaving;
-                    if (goal.Amount <= startingBalance)
+                    sumsOfAmountByImportanceAdjusted[i - 1] += amountAfterSaving;
+                    sumsOfAmountByImportanceDefault[i - 1] += entry.Amount;
+                    entryAmounts[i - 1]++;
+
+                }
+            }
+            var timesToRepeatSaving = 0;
+            var approximateSavedAmount = startingBalance;
+            while (goal.Amount > approximateSavedAmount)
+            {
+                for (var i = enumCount; i > (int)Importance.Necessary; i--)
+                {
+                    if (entryAmounts[i - 1] != 0)
                     {
-                        return true; //Saved enough
+                        averagesOfAmountByImportanceAdjusted[i - 1] = sumsOfAmountByImportanceAdjusted[i - 1] / entryAmounts[i - 1];
+                        averagesOfAmountByImportanceDefault[i - 1] = sumsOfAmountByImportanceDefault[i - 1] / entryAmounts[i - 1];
                     }
+                    else
+                    {
+                        averagesOfAmountByImportanceAdjusted[i - 1] = 0;
+                        averagesOfAmountByImportanceDefault[i - 1] = 0;
+                    }
+
+                    approximateSavedAmount += averagesOfAmountByImportanceDefault[i - 1] - averagesOfAmountByImportanceAdjusted[i - 1];
                 }
+                timesToRepeatSaving++;              
+            }
+            for (var i = enumCount; i > (int)Importance.Necessary; i--)
+            {
+                monthlySuggestions.Add(new SavingSuggestionByMonth(averagesOfAmountByImportanceAdjusted[i - 1], averagesOfAmountByImportanceDefault[i - 1], (Importance)i));
             }
```

Comment:
I remember this produce list that repeatet itself quite a few times. How was this solved?

Reply:
You're looking at the solution here, I simply ran another for loop, which has the sole purpose of generating the list, before this adding to list was above in the loop that repeated as many times, as months were needed to save